synapse/handlers/sync.py

```diff
@@ -151,6 +160,17 @@ def current_sync_callback(before_token, after_token):
             )
             defer.returnValue(result)
 
+    @defer.inlineCallbacks
+    def _is_world_readable(self, room_id):
+        state = yield self.hs.get_state_handler().get_current_state(
+            room_id,
+            EventTypes.RoomHistoryVisibility
+        )
+        if state and "history_visibility" in state.content:
+            defer.returnValue(state.content["history_visibility"] == "world_readable")
```

Comment:
I don't think this will throw because we check if the ``history_visibility`` is in the content before accessing it.
This the same check as used for ``/events`` FWIW